app/controllers/goldencobra/api/v2/articles_controller.rb

```diff
@@ -211,77 +145,84 @@ def breadcrumb
-          new_article.creator_id = current_user.id
+        def find_article
+          @article = Goldencobra::Article.where(creator_id: current_user.id,
+                                               external_referee_id: params[:referee_id]).first
 
-          if params[:article][:article_type]
-            new_article.article_type = params[:article][:article_type]
-          else
-            new_article.article_type = 'Default Show'
-          end
+          render_error("article not found", 423) and return unless @article
+        end
 
-          if params[:author].present? && params[:author][:lastname].present?
-            author = Goldencobra::Author.find_or_create_by_lastname(params[:author][:lastname])
-            new_article.author = author
-          end
+        def setup_params(old_params)
+          old_params.merge(
+            {
+              creator_id: current_user.id,
+              article_type: old_params.fetch(:article_type, "Default Show"),
+              author: author,
+              external_referee_id: params[:referee_id],
+              external_referee_ip: request.env["REMOTE_ADDR"]
+            }
+          )
+        end
 
-          if params[:images].present?
-            params[:images].each do |i|
-              img = Goldencobra::Upload.create(i[:image])
-              article.images << img
-            end
+        def author
+          if params[:author].present? && params[:author][:lastname].present?
+            Goldencobra::Author.find_or_create_by_lastname(params[:author][:lastname])
+          else
+            nil
           end
+        end
 
-
-          #Set externel Referee
-          new_article.external_referee_id = params[:referee_id]
-          new_article.external_referee_ip = request.env['REMOTE_ADDR']
-
-          # Try to save the article
+        def create_article(article_param)
+          new_params = setup_params(article_params)
+          new_article = Goldencobra::Article.new(new_params)
+          article_param.each{ |i| new_article.images << Goldencobra::Upload.create(i[:image]) }
           new_article.save
-          return new_article
+          new_article
         end
 
+        def check_for_current_user
+          render_error(nil, 403) and return unless current_user
+        end
 
-        def update_article(article_param)
-          # Input validation
-          return nil if check_presents_of_article_params(article_param) == false
-
-          # Get existing article
-          article = Goldencobra::Article.where(creator_id: current_user.id).find_by_external_referee_id(params[:referee_id])
+        def check_for_article
+          render_error("article data missing") and return unless params[:article]
+        end
 
-          # Update existing article
-          article.update_attributes(params[:article])
+        def check_for_referee
+           render_error("referee_id missing") and return unless params[:referee_id]
```

Comment:
Line is too long. [83/80]<br>Use `&&` instead of `and`.<br>Use 2 (not 3) spaces for indentation.